Source/AwsCommonRuntimeKit/http/HTTPRequest.swift

```diff
@@ -45,15 +45,38 @@ public class HTTPRequest: HTTPMessage {
                 headers: [HTTPHeader] = [HTTPHeader](),
                 body: IStreamable? = nil,
                 allocator: Allocator = defaultAllocator) throws {
-        try super.init(allocator: allocator)
+        guard let rawValue = aws_http_message_new_request(allocator.rawValue) else {
+            throw CommonRunTimeError.crtError(.makeFromLastError())
+        }
+        super.init(rawValue: rawValue, allocator: allocator)
 
         self.method = method
         self.path = path
+        self.body = body
+        addHeaders(headers: headers)
+    }
+}
 
-        if let body = body {
-            let iStreamCore = IStreamCore(iStreamable: body, allocator: allocator)
-            aws_http_message_set_body_stream(self.rawValue, &iStreamCore.rawValue)
+public class HTTP2Request: HTTPRequestBase {
+    /// Creates an http2 request which can be passed to a connection.
+    /// - Parameters:
+    ///   - headers: (Optional) headers to send
+    ///   - body: (Optional) body stream to send as part of request
+    ///   - manualDataWrites: Set it to true to indicate body data will be provided over time.
+    ///                       The data can be be supplied via `HTTP2Stream.writeData`.
+    ///                       The last data should be sent with endOfStream as true to complete the stream.
```

Comment:
`manualDataWrites` isn't supposed to be here, right?

Reply:
Good catch, this is an outdated comment.